Description: Backported fixes to manual.
 The manual in 1.2.3 contains a couple of fixes, this patch applies
 them to the 1.2.1 version. Notice that this doesn't make it identical
 to the 1.2.3 version, which also documents new features.
Author: Pietro Battiston <toobaz@email.it>
Forwarded: not-needed

--- python-shapely-1.2.1.orig/docs/manual.txt
+++ python-shapely-1.2.1/docs/manual.txt
@@ -1366,8 +1366,8 @@ Figure 1. Convex hull (blue) of 2 points
 
 .. attribute:: object.envelope
 
-  Returns a representation of the point or smallest rectangular polygon that
-  contains the object.
+  Returns a representation of the point or smallest rectangular polygon (with
+  sides parallel to the coordinate axes) that contains the object.
 
 .. sourcecode:: pycon
 
@@ -1501,7 +1501,7 @@ first use the :func:`prepared.prep` func
 .. sourcecode:: pycon
 
   >>> from shapely.geometry import Point
-  >>> from shapely.prepared import polyprep
+  >>> from shapely.prepared import prep
   >>> points = [...] # large list of points
   >>> polygon = Point(0.0, 0.0).buffer(1.0)
   >>> prepared_polygon = prep(polygon)
@@ -1514,6 +1514,26 @@ Prepared geometries instances have the f
 same arguments and usage as their counterparts in non-prepared geometric
 objects.
 
+Diagnostics
+-----------
+
+.. function:: validation.explain_validity(ob):
+
+  Returns a string explaining the validity or invalidity of the object.
+
+  `New in version 1.2.1`.
+
+The messages may or may not have a representation of a problem point that can
+be parsed out.
+
+.. sourcecode:: python
+
+  >>> coords = [(0, 0), (0, 2), (1, 1), (2, 2), (2, 0), (1, 1), (0, 0)]
+  >>> p = Polygon(coords)
+  >>> from shapely.validation import explain_validity
+  >>> explain_validity(p)
+  'Ring Self-intersection[1 1]'
+
 Interoperation
 ==============
 
